### DISCUSSION TITLE
TEAM application not loading

### DISCUSSION THREAD

**Discussion by dbrown-tiimely** [aws-samples/iam-identity-center-team#252] [Q&A] [answered]
Hi all,

I've performed a new deployment from main in the past week (and redeployed following destroy).
Deployment in a child account separate from my idc, but in the same region (ap-southeast-2).
All stages of deployment completed following the provided documentation without error, up to and including congnito.sh
On accessing the TEAM-IDC-APP application various redirects occur (as if logon is attempted), but ultimately hits the grpahic with the spinning balls (as shown in [#206](https://github.com/aws-samples/iam-identity-center-team/discussions/206), [#150](https://github.com/aws-samples/iam-identity-center-team/issues/150) and [#138)](https://github.com/aws-samples/iam-identity-center-team/discussions/138)
CloudWatch logs show the following for an attempted logon to the application.

/aws/appsync/apis/xxx

	2024-06-03T23:43:02.309Z
	{
    "logType": "ResponseMapping",
    "path": [
        "getGroups"
    ],
    "fieldName": "getGroups",
    "resolverArn": "arn:aws:appsync:ap-southeast-2:xxx:apis/xxx/types/Query/resolvers/getGroups",
    "functionName": "InvokeTeamgetGroupsLambdaDataSource",
    "requestId": "bbac4334-3f2e-4f13-95e2-a8a5f10752bb",
    "fieldInError": true,
    "errors": [
        "CustomTemplateException(message=Parameter validation failed:\nInvalid type for parameter MemberId.UserId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>, errorType=Lambda:Unhandled, data=null, errorInfo=null)"
    ],
    "parentType": "Query",
    "graphQLAPIId": "xxx",
    "functionArn": "arn:aws:appsync:ap-southeast-2:xxx:apis/xxx/functions/xxx"
	}`
	
/aws/lambda/teamgetGroups-main

	2024-06-03T23:43:01.814Z
	USER not found.
		2024-06-03T23:43:02.579Z
	LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
	LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
		2024-06-03T23:43:02.579Z
	[ERROR] ParamValidationError: Parameter validation failed:
	Invalid type for parameter MemberId.UserId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
	Traceback (most recent call last):
	  File "/var/task/index.py", line 135, in handler
		groupData = list_idc_group_membership(userId)
	  File "/var/task/index.py", line 116, in list_idc_group_membership
		for page in paginator:
	  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 269, in __iter__
		response = self._make_request(current_kwargs)
	  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 357, in _make_request
		return self._method(**current_kwargs)
	  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call
		return self._make_api_call(operation_name, kwargs)
	  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 919, in _make_api_call
		request_dict = self._convert_to_request_dict(
	  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 990, in _convert_to_request_dict
		request_dict = self._serializer.serialize_to_request(
	  File "/opt/python/lib/python3.9/site-packages/botocore/validate.py", line 381, in serialize_to_request
		raise ParamValidationError(report=report.generate_report())`

I've confirmed the changes included in PR [#151](https://github.com/aws-samples/iam-identity-center-team/pull/151) are present and deployed.
	
As this seems to have been resolved previously, i'm left wondering why i'm seeing this issue still.
It's noted in [#138](https://github.com/aws-samples/iam-identity-center-team/discussions/138) that deploying to a new AWS account resolved the issue, though i reckon this was conincidental and related to the changes in PR[#151](https://github.com/aws-samples/iam-identity-center-team/pull/151).
Any suggestion would be most welcome.

**Comment by tawoyinfa** [comment, marked as answer]
@dbrown-tiimely  the solution needs to be deployed in the same account and region as your IDC implementation

**Comment by dbrown-tiimely** [comment]
Thanks @tawoyinfa for the clarification. I had misread the documentation
TEAM app deployed and accessible.
